CAPI/cpp/API/src/logic.cpp

```diff
@@ -101,16 +101,16 @@ std::pair<int32_t, int32_t> Logic::GetConstructionState(int32_t cellX, int32_t c
     auto it3 = currentState->mapInfo->fortState.find(pos);
     if (it != currentState->mapInfo->factoryState.end())
     {
-        return currentState->mapInfo->factoryState[pos];
+        return std::make_optional<THUAI7::ConstructionState>(THUAI7::ConstructionState(currentState->mapInfo->factoryState[pos], THUAI7::ConstructionType::Factory));
```

Comment:
```suggestion
        return std::make_optional<THUAI7::ConstructionState>(currentState->mapInfo->factoryState[pos], THUAI7::ConstructionType::Factory);
```
`make_optional` 直接写构造函数的参数；显式调用构造函数会导致多一次对象的构造。